### PR TITLE
TCM-101 | Build dummy addcard screen

### PR DESCRIPTION
**Pull request by halilyesilyurt**
### Summary

- [x] Completed dummy addcard screen.
- [x] Added addcard screen navigation to test utility.

### Issue

[TCM-101](https://hyfnla.atlassian.net/browse/TCM-101)

### Video/Screenshot

add_card

https://user-images.githubusercontent.com/30659854/114233396-ec5ab600-9985-11eb-83b9-d044693614e7.mp4

### CODE REVIEW

File: app/src/main/res/layout/fragment_add_card.xml [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+        android:id="@+id/vLine"
+        android:layout_width="0dp"
+        android:layout_height="1dp"
+        android:background="@color/greyLight"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toBottomOf="@id/ivCardNumber" />
+
+    <com.trx.consumer.views.input.InputView
+        android:id="@+id/ivExpirtyDate"
```
Comment (VivekV95) [inline app/src/main/res/layout/fragment_add_card.xml:76]: There's a typo in the specs, let's change this to "ivExpiryDate"

File: app/src/main/res/layout/fragment_add_card.xml [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+                app:layout_constraintStart_toStartOf="parent"
+                app:layout_constraintTop_toTopOf="parent" />
+
+
+        </com.trx.consumer.common.CommonView>
+
+        <com.trx.consumer.common.CommonLabel
+            android:layout_width="0dp"
+            android:layout_height="wrap_content"
+            android:layout_marginStart="20dp"
```
Comment (VivekV95) [inline app/src/main/res/layout/fragment_add_card.xml:45]: Let's make this `15dp` to match the input view margins

File: app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -25,22 +28,22 @@
...
 
     <string name="inputview_placeholder_birthday">Birthday</string>
-    <string name="inputview_placeholder_card_number">Card Number</string>
+    <string name="inputview_placeholder_card_number">CREDIT CARD NUMBER</string>
     <string name="inputview_placeholder_code">Enter Code</string>
-    <string name="inputview_placeholder_cvc">CVC</string>
+    <string name="inputview_placeholder_cvv">CVV CODE</string>
     <string name="inputview_placeholder_email">Email</string>
-    <string name="inputview_placeholder_expiration">Expiration(mmyyyy)</string>
+    <string name="inputview_placeholder_expiration">EXPIRTY DATE</string>
```
Comment (VivekV95) [inline app/src/main/res/values/strings.xml:35]: Same typo from the specs, "EXPIRY"

File: app/src/main/java/com/trx/consumer/views/input/InputView.kt [hunk 4]
```diff
@@ -57,6 +58,10 @@ class InputView @JvmOverloads constructor(
                 endIconDrawable =
                     ContextCompat.getDrawable(context, R.drawable.ic_password_toggle_drawable)
             }
+            if (mState == InputViewState.CVV) {
```
Comment (VivekV95) [inline app/src/main/java/com/trx/consumer/views/input/InputView.kt:61]: Is this needed? If not, let's remove

File: app/build.gradle [hunk 5]
```diff
@@ -60,10 +60,10 @@ android {
             buildConfigField "String", "kPlatform", '"mobile"'
             buildConfigField "String", "kClassDateTimeFormat", '"yyyy-MM-dd\'T\'HH:mm:ss.SSSX"'
             buildConfigField "Integer", "kVerificationCodeLimit", "6"
-            buildConfigField "Integer", "kExpirationDateLimit", "7"
+            buildConfigField "Integer", "kExpirationDateLimit", "5"
```
Comment (VivekV95) [inline app/build.gradle:63]: @onodude I see the specs showing a 2 digit year and not 4 digit year [here](https://www.figma.com/file/wMfhxL9w3KOw6okQJtt1OD/All-Screens-(v1.0.0)?node-id=1700%3A4524), just verifying that's what we want:

File: app/src/main/java/com/trx/consumer/views/input/InputView.kt (hunk 4 above)
Comment (VivekV95) [reply app/src/main/java/com/trx/consumer/views/input/InputView.kt:61]: Ah I see it's for hiding the code. Looks like we'll need to do the for the credit card too, so let's revert this and tackle it in a separate ticket

**Review by VivekV95 — CHANGES_REQUESTED** [review]
Looks great, left a few comments

File: app/src/main/res/values/strings.xml (hunk 3 above)
Comment (VivekV95) [reply app/src/main/res/values/strings.xml:35]: Actually, let's just keep the strings consistent with what iOS has right now, we'll modify both together to match design specs is needed: [iOS](https://github.com/hyfn/trx-consumer-ios/blob/develop/TRX-Consumer/Screens/Input/InputViewState.swift)

File: app/src/main/java/com/trx/consumer/views/input/InputViewState.kt [hunk 6]
```diff
@@ -126,16 +126,6 @@ enum class InputViewState(val placeholder: Int, vararg val type: Int) {
     private fun expirationFormat(textView: TextInputEditText, before: Int, count: Int): Boolean? {
         textView.text?.let { userInput ->
             when (userInput.length) {
-                1 -> {
```
Comment (VivekV95) [inline app/src/main/java/com/trx/consumer/views/input/InputViewState.kt:129]: Any reason for the removal? If not, let's revert this

File: app/src/main/res/layout/fragment_add_card.xml (hunk 1 above)
Comment (halilyesilyurt) [reply app/src/main/res/layout/fragment_add_card.xml]: Completed.

File: app/src/main/res/layout/fragment_add_card.xml (hunk 2 above)
Comment (halilyesilyurt) [reply app/src/main/res/layout/fragment_add_card.xml]: Completed.

File: app/src/main/res/values/strings.xml (hunk 3 above)
Comment (halilyesilyurt) [reply app/src/main/res/values/strings.xml]: Completed.

File: app/src/main/java/com/trx/consumer/views/input/InputView.kt (hunk 4 above)
Comment (halilyesilyurt) [reply app/src/main/java/com/trx/consumer/views/input/InputView.kt]: Completed.

File: app/src/main/java/com/trx/consumer/views/input/InputViewState.kt (hunk 6 above)
Comment (halilyesilyurt) [reply app/src/main/java/com/trx/consumer/views/input/InputViewState.kt:129]: Yes it was needed for correct text typing.

Comment (VivekV95) [reply app/src/main/java/com/trx/consumer/views/input/InputViewState.kt:129]: This is needed in order to auto-input a "0" if any number other than "1" is typed. For ex, if a user types "2", it'll auto-input "02/". It's just a convenience thing. Either way, if we end up doing this,  let's revert and tackle in a separate ticket as it touches a core file

File: app/build.gradle (hunk 5 above)
Comment (VivekV95) [reply app/build.gradle:63]: Looks like this is still "7" on the [iOS](https://github.com/hyfn/trx-consumer-ios/blob/develop/TRX-Consumer/Core/Constants.swift#L40) side, let's revert

**Review by VivekV95 — CHANGES_REQUESTED** [review]
Couple things

File: app/build.gradle (hunk 5 above)
Comment (halilyesilyurt) [reply app/build.gradle]: Completed.

File: app/src/main/java/com/trx/consumer/views/input/InputViewState.kt (hunk 6 above)
Comment (halilyesilyurt) [reply app/src/main/java/com/trx/consumer/views/input/InputViewState.kt]: Completed.

**Review by VivekV95 — APPROVED** [review]

**Review by onodude — APPROVED** [review]

**Pull request merged**